File: backend/src/main/java/ru/nsu/brusn/smpltodo/model/entity/TaskEntity.java
package ru.nsu.brusn.smpltodo.model.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.sun.istack.NotNull;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.ZonedDateTime;

@Entity
@Setter
@Getter
@Table(name = "Tasks")
public class TaskEntity {
    @Id
    @SequenceGenerator(name = "tasks_sequence", sequenceName = "tasks_id_sequence", allocationSize = 1)
    @GeneratedValue(strategy= GenerationType.SEQUENCE, generator = "tasks_sequence")
    private Long id;

    @NotNull
    private String name;

    @NotNull
    private ZonedDateTime created;

    private ZonedDateTime startDate;
    private ZonedDateTime deadline;
    private String description;

    private Boolean important;
    private Boolean completed;

    @ManyToOne
    @JsonBackReference
    private FolderEntity folder;
}

File: smpltodo/app/src/main/java/ru/nsu/brusn/smpltodo/api/model/task/TaskEntity.kt
package ru.nsu.brusn.smpltodo.api.model.task

import java.io.Serializable
import java.time.ZonedDateTime

class TaskEntity (
    val id: Long,
    var name: String,
    var created: ZonedDateTime,
    var startDate: ZonedDateTime?,
    var deadline: ZonedDateTime?,
    var description: String?,
    var important: Boolean,
    var completed: Boolean,
): Serializable